kratos/solving_strategies/strategies/butcher_tableau.h

```diff
@@ -120,12 +122,19 @@ class ButcherTableau
      *          intended to be used with std::inner_product.
      */
 
-    ArraySlice GetMatrixRow(const unsigned int SubStepIndex) const
+    std::tuple<RowType::const_iterator, RowType::const_iterator> GetMatrixRow(const unsigned int SubStepIndex) const
     {
-        return ArraySlice{
-            mA[SubStepIndex - 1].begin(),
-            mA[SubStepIndex - 1].begin() + SubStepIndex
-        };
+        return {mA[SubStepIndex - 1].begin(), mA[SubStepIndex - 1].begin() + SubStepIndex};
+    }
+
+    RowType::const_iterator GetMatrixRowBegin(const unsigned int SubStepIndex) const
```

Comment:
```suggestion
    RowType::const_iterator GetMatrixRowBegin(const unsigned int SubStepIndex) const noexcept
```

This is not really done in Kratos but I guess we have to start somewhere.
It's going to have negligible performance impact here, but it's more a good practices kinda thing.